embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/ndk/EmbraceNdkServiceTest.kt

```diff
@@ -214,12 +228,13 @@ internal class EmbraceNdkServiceTest {
 
     @Test
     fun `test initialization with unity id and ndk enabled runs installSignals and updateDeviceMetaData`() {
-        every { Uuid.getEmbUuid() } returns "unityId"
+        val unityId = "unityId"
+        every { Uuid.getEmbUuid() } returns unityId
         enableNdk(true)
-        every { sharedObjectLoader.loadEmbraceNative() } returns true
+
         appFramework = Embrace.AppFramework.UNITY
         initializeService()
-        assertTrue(activityService.listeners.contains(embraceNdkService))
+        assertEquals(1, activityService.listeners.size)
```

Comment:
Making embraceNdkService a spy means the object references won't match. Changed to verify that a listener exists